src/main/java/org/hyperskill/hstest/stage/SpringTest.java

```diff
@@ -135,6 +154,22 @@ public void tearDown() {
 
     public static void startSpring() throws Exception {
         if (!springRunning) {
+            String annotationPath = "org.springframework.boot.autoconfigure.SpringBootApplication";
+            List<Class<?>> suitableClasses = ReflectionUtils.getTypesAnnotatedWith(annotationPath);
+
+            int length = suitableClasses.size();
+            if (length == 0) {
+                throw new UnexpectedError("No class found with annotation " + annotationPath);
```

Comment:
That's the user error that they didn't write class with such annotation. `UnexpectedError` is reserved for errors in tests and for errors in testing library. For user errors, it's better to throw `ErrorWithFeedback`

Reply:
Also, since this error text will be shown to the learner (beginner programmer) it's better to show more user-friendly feedback. You can add `, please write this annotation in a class with your main method` to tell user what to do exactly.